From 8b026fe1bf52d9ee87d0ca49f263833d4b9b7d32 Mon Sep 17 00:00:00 2001
From: Heinrich Toews <heinrich.toews@wago.com>
Date: Thu, 19 Sep 2019 11:42:28 +0200
Subject: [PATCH] dts: 768_3301: add new hw descr for devconf 3005

This commit adds support for the new slp00001773.002 circuit board.

We agreed to offer a new devconf device id to able to support old hardware.

devconf for old hardware (psm - passive serial mode): 0x1005
devconf for the hardware (asm - active serial mode): 0x3005

Signed-off-by: Heinrich Toews <heinrich.toews@wago.com>
---
 arch/arm/boot/dts/Makefile                     |   3 +-
 arch/arm/boot/dts/am335x-pac-768_3301.dts      | 117 +++++++
 arch/arm/boot/dts/am335x-pac-768_3301_psm.dts  |   2 +-
 arch/arm/boot/dts/am335x-pac-768_330x.dtsi     |  98 +++---
 arch/arm/boot/dts/am335x-pac-768_330x_psm.dtsi | 457 +++++++++++++++++++++++++
 arch/arm/boot/dts/am335x-pfc.dtsi              |   2 +-
 6 files changed, 636 insertions(+), 43 deletions(-)
 create mode 100644 arch/arm/boot/dts/am335x-pac-768_3301.dts
 create mode 100644 arch/arm/boot/dts/am335x-pac-768_330x_psm.dtsi

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index ef6c0d3..76bfdda 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -582,7 +582,8 @@ dtb-$(CONFIG_SOC_AM33XX) += \
 	am335x-pfc100-setupfw.dtb \
 	am335x-pfc200v2-setupfw.dtb \
 	am335x-pfc200v3-setupfw.dtb \
-	am335x-pac-768_3301.dtb
+	am335x-pac-768_3301.dtb \
+	am335x-pac-768_3301_psm.dtb
 dtb-$(CONFIG_ARCH_OMAP4) += \
 	omap4-duovero-parlor.dtb \
 	omap4-kc1.dtb \
diff --git a/arch/arm/boot/dts/am335x-pac-768_3301.dts b/arch/arm/boot/dts/am335x-pac-768_3301.dts
new file mode 100644
index 0000000..bdb1b52
--- /dev/null
+++ b/arch/arm/boot/dts/am335x-pac-768_3301.dts
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2019 WAGO Kontakttechnik GmbH & Co. KG - https://www.wago.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+/dts-v1/;
+
+#include "am335x-pac-768_330x.dtsi"
+
+/{
+	model = "WAGO PAC100 768-3301";
+	compatible = "wago,am335x-pac-768_330x-3005", "wago,am335x-pfc", "ti,am33xx";
+};
+
+&encsw {
+	status ="okay";
+};
+
+&encsw {
+	status ="okay";
+};
+
+&aes {
+	status = "okay";
+};
+
+&cppi41dma {
+	status = "okay";
+};
+
+&elm {
+	status = "okay";
+};
+
+&oms {
+	status = "okay";
+};
+
+&oms_reset_all {
+	gpios = <&gpio3 15 GPIO_ACTIVE_LOW>;
+};
+
+&oms_stop {
+	gpios = <&gpio2 23 GPIO_ACTIVE_LOW>;
+};
+
+&sham {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&usb {
+	status = "okay";
+};
+
+&usb_ctrl_mod {
+	status = "okay";
+};
+
+&usb0_phy {
+	status = "okay";
+};
+
+&usb0 {
+	dr_mode = "peripheral";
+	status = "okay";
+};
+
+&spi1 {
+	status = "okay";
+};
+
+&watchdog {
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+};
+
+&swcfg_mv88e6321 {
+	status = "okay";
+};
+
+&mac {
+	status = "okay";
+};
+
+&mmc1 {
+	status = "okay";
+};
+
+&mmc2 {
+	status = "okay";
+};
+
+&gpmc {
+	status = "okay";
+};
+
+&rmd {
+	status = "okay";
+};
+
+&uio_rmd_irq0 {
+	status = "okay";
+};
+
diff --git a/arch/arm/boot/dts/am335x-pac-768_3301_psm.dts b/arch/arm/boot/dts/am335x-pac-768_3301_psm.dts
index 2aaa282..85603fd 100644
--- a/arch/arm/boot/dts/am335x-pac-768_3301_psm.dts
+++ b/arch/arm/boot/dts/am335x-pac-768_3301_psm.dts
@@ -7,7 +7,7 @@
  */
 /dts-v1/;
 
-#include "am335x-pac-768_330x.dtsi"
+#include "am335x-pac-768_330x_psm.dtsi"
 
 /{
 	model = "WAGO PAC100 768-3301";
diff --git a/arch/arm/boot/dts/am335x-pac-768_330x.dtsi b/arch/arm/boot/dts/am335x-pac-768_330x.dtsi
index 5ee4c41..20a1ce9 100644
--- a/arch/arm/boot/dts/am335x-pac-768_330x.dtsi
+++ b/arch/arm/boot/dts/am335x-pac-768_330x.dtsi
@@ -93,7 +93,9 @@
 &am33xx_pinmux {
 	pinctrl-names = "default";
 	pinctrl-0 = <
+		&gpio0_pins
 		&gpio1_pins
+		&gpio2_pins
 	>;
 
 	watchdog_pins: pinmux_watchdog_pins {
@@ -111,6 +113,15 @@
 		>;
 	};
 
+	spi1_sleep_pins: pinmux_spi1_sleep_pins {
+		pinctrl-single,pins = <
+			AM33XX_UART0_CTSN_GPIO1_8(PIN_INPUT_PULLDOWN)
+			AM33XX_UART0_RTSN_GPIO1_9(PIN_INPUT_PULLDOWN)
+			AM33XX_MCASP0_ACLKX_GPIO3_14(PIN_INPUT_PULLDOWN)
+			AM33XX_ECAP0_IN_PWM0_OUT_GPIO0_7(PIN_INPUT_PULLDOWN)
+		>;
+	};
+
 	mmc1_cd_pins: pinmux_sdcard_cd_pins {
 		pinctrl-single,pins = <
 			AM33XX_GPMC_A3_GPIO1_19(PIN_INPUT_PULLUP)
@@ -154,9 +165,9 @@
 		>;
 	};
 
-	rmii1_pins: pinmux_rmii1_pins {
+	rmii1_sleep_pins: pinmux_rmii1_sleep_pins {
 		pinctrl-single,pins = <
-			/* RMII1 could used on Marvell based devices only; default off*/
+			/* all pins are hard-wired to GROUND */
 			AM33XX_RMII1_REF_CLK_GPIO0_29(PIN_INPUT)
 			AM33XX_MII1_TX_EN_GPIO3_3(PIN_INPUT)
 			AM33XX_MII1_TXD0_GPIO0_28(PIN_INPUT)
@@ -164,25 +175,29 @@
 			AM33XX_MII1_RXD0_GPIO2_21(PIN_INPUT)
 			AM33XX_MII1_RXD1_GPIO2_20(PIN_INPUT)
 			AM33XX_MII1_CRS_GPIO3_1(PIN_INPUT)
+			AM33XX_MII1_RX_ER_GPIO3_2(PIN_INPUT)
 		>;
 	};
 
-	rmii1_sleep_pins: pinmux_rmii1_sleep_pins {
+	gpio2_pins: pinmux_gpio2_pins {
 		pinctrl-single,pins = <
-			/* RMII1 could used on Marvell based devices only; default off*/
-			AM33XX_RMII1_REF_CLK_GPIO0_29(PIN_INPUT)
-			AM33XX_MII1_TX_EN_GPIO3_3(PIN_INPUT)
-			AM33XX_MII1_TXD0_GPIO0_28(PIN_INPUT)
-			AM33XX_MII1_TXD1_GPIO0_21(PIN_INPUT)
-			AM33XX_MII1_RXD0_GPIO2_21(PIN_INPUT)
-			AM33XX_MII1_RXD1_GPIO2_20(PIN_INPUT)
-			AM33XX_MII1_CRS_GPIO3_1(PIN_INPUT)
+			AM33XX_LCD_HSYNC_GPIO2_23(PIN_INPUT) /* BAS Stop */
 		>;
 	};
 
 	gpio1_pins: pinmux_gpio1_pins {
 		pinctrl-single,pins = <
 			AM33XX_GPMC_A2_GPIO1_18(PIN_OUTPUT_PULLUP | SLEWCTRL_SLOW)
+			AM33XX_GPMC_A8_GPIO1_24(PIN_INPUT) /* DIAG-UF1 */
+			AM33XX_GPMC_A1_GPIO1_17(PIN_INPUT) /* DIAG-US1 */
+			AM33XX_GPMC_A9_GPIO1_25(PIN_INPUT) /* nINT-DIP */
+		>;
+	};
+
+	gpio0_pins: pinmux_gpio0_pins {
+		pinctrl-single,pins = <
+			AM33XX_XDMA_EVENT_INTR0_GPIO0_19(PIN_INPUT) /* nRST-LEDServer */
+			AM33XX_GPMC_WPN_GPIO0_31(PIN_INPUT) /* nINT-MCU-CC */
 		>;
 	};
 
@@ -210,7 +225,6 @@
 	gpmc_addr_pins: pinmux_gpmc_addr_pins {
 		pinctrl-single,pins = <
 			AM33XX_LCD_VSYNC_GPMC_A8(PIN_OUTPUT_PULLDOWN | SLEWCTRL_SLOW)
-			AM33XX_LCD_HSYNC_GPMC_A9(PIN_OUTPUT_PULLDOWN | SLEWCTRL_SLOW)
 		>;
 	};
 
@@ -228,19 +242,28 @@
 
 	fpga_pins: pinmux_fpga_pins {
 		pinctrl-single,pins = <
-			AM33XX_MII1_TXD3_GPIO0_16(PIN_OUTPUT_PULLUP)
-			AM33XX_LCD_DATA10_GPIO2_16(PIN_INPUT)
-			AM33XX_LCD_PCLK_GPIO2_24(PIN_INPUT)
-			AM33XX_LCD_DATA11_GPIO2_17(PIN_OUTPUT)
+			AM33XX_MII1_TXD3_GPIO0_16(PIN_OUTPUT_PULLUP) /* nCONFIG */
+			AM33XX_LCD_DATA10_GPIO2_16(PIN_INPUT) /* CONFIG-DONE */
+			AM33XX_LCD_PCLK_GPIO2_24(PIN_INPUT) /* nSTATUS */
+			AM33XX_LCD_DATA11_GPIO2_17(PIN_OUTPUT) /* nRST */
+			AM33XX_GPMC_WAIT0_GPIO0_30(PIN_OUTPUT) /* nCE */
+			AM33XX_MII1_TXD2_GPIO0_17(PIN_INPUT) /* INIT-DONE */
 		>;
 	};
 
 	rmd_pins: pinmux_rmd_pins {
 		pinctrl-single,pins = <
-			AM33XX_LCD_DATA6_GPIO2_12(PIN_INPUT)
-			AM33XX_LCD_AC_BIAS_EN_GPIO2_25(PIN_INPUT)
-			AM33XX_XDMA_EVENT_INTR1(PIN_INPUT)
-			AM33XX_SPI0_CS1_XDMA_EVENT_INTR2(PIN_INPUT)
+			AM33XX_GPMC_A7_GPIO1_23(PIN_INPUT) /* DBG0 */
+			AM33XX_GPMC_A6_GPIO1_22(PIN_INPUT) /* DBG3 */
+
+			AM33XX_LCD_DATA7_GPIO2_13(PIN_INPUT) /* SYNC1 */
+			AM33XX_LCD_DATA8_GPIO2_14(PIN_INPUT) /* SYNC2 */
+			AM33XX_LCD_DATA9_GPIO2_15(PIN_INPUT) /* SYNC3 */
+
+			AM33XX_LCD_DATA6_GPIO2_12(PIN_INPUT) /* DMA2-RMD */
+			AM33XX_LCD_AC_BIAS_EN_GPIO2_25(PIN_INPUT) /* IRQ1-RMD */
+			AM33XX_XDMA_EVENT_INTR1(PIN_INPUT) /* DMA1-RMD */
+			AM33XX_SPI0_CS1_XDMA_EVENT_INTR2(PIN_INPUT) /* DMA0-RMD */
 		>;
 	};
 };
@@ -252,24 +275,6 @@
 };
 
 &spi1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&spi1_pins>;
-	ti,pindir-d0-out-d1-in;
-
-	fpga_ps: fpga_ps@0 {
-		compatible = "altr,passive-serial";
-		pinctrl-names = "default";
-		pinctrl-0 = <&fpga_pins>;
-
-		reg = <0>;
-		spi-max-frequency = <48000000>;
-
-		reset-gpios = <&gpio2 17 GPIO_ACTIVE_LOW>;
-
-		gpio-nconfig = <&gpio0 16 GPIO_ACTIVE_LOW>;
-		gpio-nstatus = <&gpio2 24 GPIO_ACTIVE_LOW>;
-		gpio-confdone = <&gpio2 16 GPIO_ACTIVE_LOW>;
-	};
 };
 
 &watchdog {
@@ -312,6 +317,20 @@
 		#gpio-cells = <2>;
 		gpio-controller;
         };
+
+	/*
+	 * PTN5150H
+	 * CC logic for USB Type-C applications
+	 * yet no driver configured!
+	 *
+	 * Initially functions in device only mode.
+	 * But it's possible to enable both directions.
+	 *
+	 * Pins: nRST-CC (&io_expander_70 6)  and nINT-CC
+	 */
+	cc_logic_usb_c: cclogic@3c {
+		reg = <0x3c>;
+	};
 };
 
 /delete-node/ &led_bar61;
@@ -376,12 +395,11 @@
 &mac {
 	pinctrl-0 = <
 		&rmii2_pins
-		&rmii1_pins
+		&rmii1_sleep_pins
 	>;
 
 	pinctrl-1 = <
 		&rmii2_sleep_pins
-		&rmii1_sleep_pins
 	>;
 };
 
diff --git a/arch/arm/boot/dts/am335x-pac-768_330x_psm.dtsi b/arch/arm/boot/dts/am335x-pac-768_330x_psm.dtsi
new file mode 100644
index 0000000..6406b59
--- /dev/null
+++ b/arch/arm/boot/dts/am335x-pac-768_330x_psm.dtsi
@@ -0,0 +1,457 @@
+/*
+ * Copyright (C) 2019 WAGO Kontakttechnik GmbH & Co. KG - https://www.wago.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "am335x-pfc.dtsi"
+#include "am335x-pfc-750_8xxx-leds.dtsi"
+#include "am335x-pfc-750_821x-mv88e6321.dtsi"
+#include "wago-devconf.dtsi"
+
+/ {
+	memory {
+		device_type = "memory";
+		reg = <0x80000000 0x20000000>; /* 512 MB */
+	};
+
+	uio_rmd_mem@1000000 {
+		compatible = "uio_pdrv_genirq";
+		reg = <0x01000000 0x01000000>;
+	};
+
+	/* remove this node if interrupt is used in rmd driver and inform the
+	 * rlb team to use the rmd_irq1 node for error handling.
+	 */
+	uio_rmd_irq0: uio_rmd_irq0 {
+		status = "disabled";
+		compatible = "uio_pdrv_genirq";
+		interrupts = <7>;
+		interrupt-parent = <&intc>;
+	};
+
+	uio_rmd_irq1: uio_rmd_irq1 {
+		status = "disabled";
+		compatible = "uio_pdrv_genirq";
+		interrupts = <25 IRQ_TYPE_EDGE_RISING>;
+		interrupt-parent = <&gpio2>;
+	};
+
+	rmd: rmd {
+		status = "disabled";
+		compatible = "wago,rmd";
+		pinctrl-names = "default";
+		pinctrl-0 = <&rmd_pins>;
+
+		rmd,fifo-size = <0x1000>;
+		reg = <0x01000000 0x01000000>;
+
+		/*
+		 * Crossbar Mapped Channels
+		 *
+		 * 28: pi_x_dma_event_intr0
+		 * 29: pi_x_dma_event_intr1
+		 *
+		 * Map 28 and 29 to Open Channels 20 and 21
+		 *
+		 */
+		dmas = <&edma_xbar 12 0 30>, <&edma_xbar 13 0 29>;
+		dma-names = "rx", "tx";
+
+		interrupts = <12 IRQ_TYPE_EDGE_RISING>;
+		interrupt-parent = <&gpio2>;
+	};
+
+	encsw: dip-switch {
+		status = "disabled";
+		compatible = "encsw";
+		encsw-gpios = <
+			&io_expander_71 0 GPIO_ACTIVE_LOW
+			&io_expander_71 1 GPIO_ACTIVE_LOW
+			&io_expander_71 2 GPIO_ACTIVE_LOW
+			&io_expander_71 3 GPIO_ACTIVE_LOW
+			&io_expander_71 4 GPIO_ACTIVE_LOW
+			&io_expander_71 5 GPIO_ACTIVE_LOW
+			&io_expander_71 6 GPIO_ACTIVE_LOW
+			&io_expander_71 7 GPIO_ACTIVE_LOW
+			>;
+	};
+};
+
+&cpu_0 {
+	operating-points = <
+		1000000  1325000
+		 800000  1260000
+		 720000  1200000
+		 600000  1100000
+		 300000  950000
+	>;
+};
+
+&am33xx_pinmux {
+	pinctrl-names = "default";
+	pinctrl-0 = <
+		&gpio1_pins
+	>;
+
+	watchdog_pins: pinmux_watchdog_pins {
+		pinctrl-single,pins = <
+			AM33XX_EMU1_GPIO3_8(PIN_OUTPUT_PULLDOWN)
+		>;
+	};
+
+	spi1_pins: pinmux_spi1_pins {
+		pinctrl-single,pins = <
+			AM33XX_UART0_CTSN_SPI1_D0(PIN_INPUT)
+			AM33XX_UART0_RTSN_SPI1_D1(PIN_OUTPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_MCASP0_ACLKX_SPI1_SCLK(PIN_INPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_ECAP0_IN_PWM0_OUT_SPI1_CS1(PIN_OUTPUT_PULLUP | SLEWCTRL_SLOW)
+		>;
+	};
+
+	mmc1_cd_pins: pinmux_sdcard_cd_pins {
+		pinctrl-single,pins = <
+			AM33XX_GPMC_A3_GPIO1_19(PIN_INPUT_PULLUP)
+		>;
+	};
+
+	emmc_pins: pinmux_emmc_pins {
+		pinctrl-single,pins = <
+			AM33XX_GPMC_CSN1_MMC1_CLK(PIN_INPUT_PULLUP | SLEWCTRL_FAST)
+			AM33XX_GPMC_CSN2_MMC1_CMD(PIN_INPUT_PULLUP | SLEWCTRL_FAST)
+			AM33XX_MII1_TX_CLK_MMC1_DAT0(PIN_INPUT_PULLUP | SLEWCTRL_FAST)
+			AM33XX_MII1_RX_CLK_MMC1_DAT1(PIN_INPUT_PULLUP | SLEWCTRL_FAST)
+			AM33XX_MII1_RXD3_MMC1_DAT2(PIN_INPUT_PULLUP | SLEWCTRL_FAST)
+			AM33XX_MII1_RXD2_MMC1_DAT3(PIN_INPUT_PULLUP | SLEWCTRL_FAST)
+		>;
+	};
+
+	rmii2_pins: pinmux_rmii2_pins {
+		pinctrl-single,pins = <
+			/* RMII 2 */
+			AM33XX_GPMC_A0_RMII2_TXEN(PIN_OUTPUT | SLEWCTRL_FAST)
+			AM33XX_GPMC_A4_RMII2_TXD1(PIN_OUTPUT | SLEWCTRL_FAST)
+			AM33XX_GPMC_A5_RMII2_TXD0(PIN_OUTPUT | SLEWCTRL_FAST)
+			AM33XX_GPMC_A10_RMII2_RXD1(PIN_INPUT)
+			AM33XX_GPMC_A11_RMII2_RXD0(PIN_INPUT)
+			AM33XX_MII1_COL_RMII2_REFCLK(PIN_INPUT)
+			AM33XX_GPMC_CSN3_RMII2_CRS_DV(PIN_INPUT)
+		>;
+	};
+
+	rmii2_sleep_pins: pinmux_rmii2_sleep_pins {
+		pinctrl-single,pins = <
+			/* RMII 2 */
+			AM33XX_GPMC_A0_RMII2_TXEN(PIN_OUTPUT | SLEWCTRL_FAST)
+			AM33XX_GPMC_A4_RMII2_TXD1(PIN_OUTPUT | SLEWCTRL_FAST)
+			AM33XX_GPMC_A5_RMII2_TXD0(PIN_OUTPUT | SLEWCTRL_FAST)
+			AM33XX_GPMC_A10_RMII2_RXD1(PIN_INPUT)
+			AM33XX_GPMC_A11_RMII2_RXD0(PIN_INPUT)
+			AM33XX_MII1_COL_RMII2_REFCLK(PIN_INPUT)
+			AM33XX_GPMC_CSN3_RMII2_CRS_DV(PIN_INPUT)
+		>;
+	};
+
+	rmii1_pins: pinmux_rmii1_pins {
+		pinctrl-single,pins = <
+			/* RMII1 could used on Marvell based devices only; default off*/
+			AM33XX_RMII1_REF_CLK_GPIO0_29(PIN_INPUT)
+			AM33XX_MII1_TX_EN_GPIO3_3(PIN_INPUT)
+			AM33XX_MII1_TXD0_GPIO0_28(PIN_INPUT)
+			AM33XX_MII1_TXD1_GPIO0_21(PIN_INPUT)
+			AM33XX_MII1_RXD0_GPIO2_21(PIN_INPUT)
+			AM33XX_MII1_RXD1_GPIO2_20(PIN_INPUT)
+			AM33XX_MII1_CRS_GPIO3_1(PIN_INPUT)
+		>;
+	};
+
+	rmii1_sleep_pins: pinmux_rmii1_sleep_pins {
+		pinctrl-single,pins = <
+			/* RMII1 could used on Marvell based devices only; default off*/
+			AM33XX_RMII1_REF_CLK_GPIO0_29(PIN_INPUT)
+			AM33XX_MII1_TX_EN_GPIO3_3(PIN_INPUT)
+			AM33XX_MII1_TXD0_GPIO0_28(PIN_INPUT)
+			AM33XX_MII1_TXD1_GPIO0_21(PIN_INPUT)
+			AM33XX_MII1_RXD0_GPIO2_21(PIN_INPUT)
+			AM33XX_MII1_RXD1_GPIO2_20(PIN_INPUT)
+			AM33XX_MII1_CRS_GPIO3_1(PIN_INPUT)
+		>;
+	};
+
+	gpio1_pins: pinmux_gpio1_pins {
+		pinctrl-single,pins = <
+			AM33XX_GPMC_A2_GPIO1_18(PIN_OUTPUT_PULLUP | SLEWCTRL_SLOW)
+		>;
+	};
+
+	gpmc_data_pins: pinmux_gpmc_data_pins {
+		pinctrl-single,pins = <
+			AM33XX_GPMC_AD0(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD1(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD2(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD3(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD4(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD5(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD6(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD7(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD8(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD9(PIN_INPUT_PULLDOWN  | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD10(PIN_INPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD11(PIN_INPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD12(PIN_INPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD13(PIN_INPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD14(PIN_INPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_GPMC_AD15(PIN_INPUT_PULLDOWN | SLEWCTRL_SLOW)
+		>;
+	};
+
+	gpmc_addr_pins: pinmux_gpmc_addr_pins {
+		pinctrl-single,pins = <
+			AM33XX_LCD_VSYNC_GPMC_A8(PIN_OUTPUT_PULLDOWN | SLEWCTRL_SLOW)
+			AM33XX_LCD_HSYNC_GPMC_A9(PIN_OUTPUT_PULLDOWN | SLEWCTRL_SLOW)
+		>;
+	};
+
+	gpmc_ctrl_pins: pinmux_gpmc_ctrl_pins {
+		pinctrl-single,pins = <
+			AM33XX_GPMC_OEN_REN(PIN_OUTPUT_PULLUP  | SLEWCTRL_FAST)
+			AM33XX_GPMC_WEN(PIN_OUTPUT_PULLUP      | SLEWCTRL_FAST)
+			AM33XX_GPMC_CSN0(PIN_OUTPUT_PULLUP     | SLEWCTRL_SLOW)
+			AM33XX_GPMC_BEN0_CLE(PIN_OUTPUT_PULLUP | SLEWCTRL_FAST)
+			AM33XX_GPMC_BEN1(PIN_OUTPUT_PULLUP     | SLEWCTRL_FAST)
+			AM33XX_GPMC_CLK_GPMC_WAIT1(PIN_INPUT_PULLUP)
+			AM33XX_GPMC_ADVN_ALE(PIN_OUTPUT)
+		>;
+	};
+
+	fpga_pins: pinmux_fpga_pins {
+		pinctrl-single,pins = <
+			AM33XX_MII1_TXD3_GPIO0_16(PIN_OUTPUT_PULLUP)
+			AM33XX_LCD_DATA10_GPIO2_16(PIN_INPUT)
+			AM33XX_LCD_PCLK_GPIO2_24(PIN_INPUT)
+			AM33XX_LCD_DATA11_GPIO2_17(PIN_OUTPUT)
+		>;
+	};
+
+	rmd_pins: pinmux_rmd_pins {
+		pinctrl-single,pins = <
+			AM33XX_LCD_DATA6_GPIO2_12(PIN_INPUT)
+			AM33XX_LCD_AC_BIAS_EN_GPIO2_25(PIN_INPUT)
+			AM33XX_XDMA_EVENT_INTR1(PIN_INPUT)
+			AM33XX_SPI0_CS1_XDMA_EVENT_INTR2(PIN_INPUT)
+		>;
+	};
+};
+
+&wsysinit {
+	board,variant = "pac100";
+	tty,rs232-485 = "nop";
+	tty,service   = "ttyGS0";
+};
+
+&spi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi1_pins>;
+	ti,pindir-d0-out-d1-in;
+
+	fpga_ps: fpga_ps@0 {
+		compatible = "altr,passive-serial";
+		pinctrl-names = "default";
+		pinctrl-0 = <&fpga_pins>;
+
+		reg = <0>;
+		spi-max-frequency = <48000000>;
+
+		reset-gpios = <&gpio2 17 GPIO_ACTIVE_LOW>;
+
+		gpio-nconfig = <&gpio0 16 GPIO_ACTIVE_LOW>;
+		gpio-nstatus = <&gpio2 24 GPIO_ACTIVE_LOW>;
+		gpio-confdone = <&gpio2 16 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&watchdog {
+	pinctrl-names = "default";
+	pinctrl-0 = <&watchdog_pins>;
+
+	gpio-en = <&io_expander_70 0 GPIO_ACTIVE_LOW>;
+	gpios = <&gpio3 8 GPIO_ACTIVE_LOW>;
+};
+
+&mv88e6321_switch {
+	ports {
+		port@3 {
+			label = "ethX1";
+		};
+
+		port@4 {
+			label = "ethX2";
+		};
+	};
+};
+
+&i2c0 {
+	/* IO Port Expander outputs only */
+	io_expander_70: pca9538@70 {
+		compatible = "nxp,pca9538";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		#gpio-cells = <2>;
+		gpio-controller;
+		reg = <0x70>;
+	};
+
+	/* IO Port Expander for DIP-switch, inputs only */
+	io_expander_71: pca9538@71 {
+		compatible = "nxp,pca9538";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x71>;
+		#gpio-cells = <2>;
+		gpio-controller;
+        };
+};
+
+/delete-node/ &led_bar61;
+
+&lb60_0 {
+	label = "ds-red";
+};
+
+&lb60_1 {
+	label = "ds-green";
+};
+
+&lb60_2 {
+	label = "ec-red";
+};
+
+&lb60_3 {
+	label = "ec-blue";
+};
+
+&lb60_4 {
+	label = "ms-red";
+};
+
+&lb60_5 {
+	label = "ms-green";
+};
+
+&lb60_6 {
+	label = "ns-red";
+};
+
+&lb60_7 {
+	label = "ns-green";
+};
+
+&lb60_8 {
+	label = "run-red";
+};
+
+&lb60_9 {
+	label = "run-green";
+};
+
+&lb60_10 {
+	label = "app-red";
+};
+
+&lb60_11 {
+	label = "app-green";
+};
+
+&rtc_i2c {
+	trim-data = <0 1 31>;
+};
+
+&tps {
+	interrupt-parent = <&gpio3>;
+	interrupts = <20 IRQ_TYPE_LEVEL_LOW>;
+};
+
+&mac {
+	pinctrl-0 = <
+		&rmii2_pins
+		&rmii1_pins
+	>;
+
+	pinctrl-1 = <
+		&rmii2_sleep_pins
+		&rmii1_sleep_pins
+	>;
+};
+
+&mmc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <
+		&mmc1_pins
+		&mmc1_cd_pins
+	>;
+
+	cd-gpios = <&gpio1 19 GPIO_ACTIVE_LOW>;
+};
+
+&mmc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&emmc_pins>;
+	vmmc-supply = <&dcdc4_reg>;
+	bus-width = <4>;
+	ti,non-removable;
+};
+
+&gpio1 {
+	boot_select {
+		gpio-hog;
+		gpios = <18 GPIO_ACTIVE_LOW>;
+		output-high;
+		line-name = "boot_select";
+	};
+};
+
+&gpmc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&gpmc_data_pins>, <&gpmc_addr_pins>, <&gpmc_ctrl_pins>;
+
+	ranges = <0 0 0x01000000 0x01000000>;
+
+	fpga: fpga@0,0 {
+		reg = <0 0 0x01000000>;
+		bank-width = <2>; /* 16 Bit */
+
+		gpmc,sync-clk-ps = <0>;
+		gpmc,cs-on-ns = <0>;
+		gpmc,cs-rd-off-ns = <80>;
+		gpmc,cs-wr-off-ns = <70>;
+
+		gpmc,adv-on-ns = <0>;
+		gpmc,adv-rd-off-ns = <10>;
+		gpmc,adv-wr-off-ns = <10>;
+
+		gpmc,we-on-ns = <20>;
+		gpmc,we-off-ns = <70>;
+		gpmc,oe-on-ns = <10>;
+		gpmc,oe-off-ns = <80>;
+		gpmc,oe-extra-delay;
+
+		gpmc,access-ns = <80>;
+		gpmc,rd-cycle-ns = <100>;
+		gpmc,wr-cycle-ns = <90>;
+		gpmc,page-burst-access-ns = <0>;
+
+		gpmc,bus-turnaround-ns  = <0>;
+		gpmc,cycle2cycle-diffcsen;
+		gpmc,cycle2cycle-samecsen;
+		gpmc,cycle2cycle-delay-ns = <10>;
+		gpmc,wr-access-ns = <70>;
+		gpmc,wr-data-mux-bus-ns = <20>;
+
+		gpmc,wait-pin = <1>;
+		gpmc,wait-on-read;
+		gpmc,wait-on-write;
+		gpmc,mux-add-data = <2>; /* address-data multiplexing */
+	};
+};
diff --git a/arch/arm/boot/dts/am335x-pfc.dtsi b/arch/arm/boot/dts/am335x-pfc.dtsi
index 690f4eb..7552555 100644
--- a/arch/arm/boot/dts/am335x-pfc.dtsi
+++ b/arch/arm/boot/dts/am335x-pfc.dtsi
@@ -69,7 +69,7 @@
 			debounce-interval = <1>;
 		};
 
-		stop {
+		oms_stop: stop {
 			label = "STOP";
 			gpios = <&gpio3 17 GPIO_ACTIVE_LOW>;
 			linux,code = <2>;
-- 
2.7.4

